python/katana/local/_graph.pyx

```diff
@@ -481,6 +531,182 @@ cdef class GraphBase:
+
+    def get_node_entity_type_id(self, str name):
+        """
+        Return the EntityTypeID for an atomic type with `name` for nodes
+        Return None if `name` is not an atomic node type
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        cdef string input_str = bytes(name, "utf-8")
+        if not manager.HasAtomicType(input_str):
+            return None
+        return manager.GetEntityTypeID(input_str)
+
+    def get_edge_entity_type_id(self, str name):
+        """
+        Return the EntityTypeID for an atomic type with `name` for edges
+        Return None if `name` is not an atomic edge type
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        cdef string input_str = bytes(name, "utf-8")
+        if not manager.HasAtomicType(input_str):
+            return None
+        return manager.GetEntityTypeID(input_str)
+
+    def node_entity_type_to_type_name_set(self, entity_type_id: int):
+        """
+        Return a set of strings representing the names of the types for the type ID
+        `entity_type_id` for nodes
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        return handle_result_TypeNameSet(manager.EntityTypeToTypeNameSet(entity_type_id))
+
+    def edge_entity_type_to_type_name_set(self, entity_type_id: int):
+        """
+        Return a set of strings representing the names of the types for the type ID
+        `entity_type_id` for edges
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        return handle_result_TypeNameSet(manager.EntityTypeToTypeNameSet(entity_type_id))
+
+    def get_atomic_node_type_name(self, entity_type_id: int):
+        """
+        Return the type name if `entity_type_id` is a valid node entity type ID. Otherwise,
+        return None
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        r =  manager.GetAtomicTypeName(entity_type_id)
+        if r:
+            return r.value()
+        return None
+
+    def get_atomic_edge_type_name(self, entity_type_id: int):
+        """
+        Return the type name if `entity_type_id` is a valid edge entity type ID. Otherwise,
+        return None
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        r =  manager.GetAtomicTypeName(entity_type_id)
+        if r:
+            return r.value()
+        return None
+
+    def get_atomic_node_entity_type_ids(self):
+        """
+        Return a list of all node entity type IDs
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        return manager.GetAtomicEntityTypeIDs()
```

Comment:
I think all the "get list" methods can be combined into a two properties one returning a dict of ID to `EntityType` objects, and the other returning a dict of `str` (name) to `EntityType`. If someone wants the EntityTypes only they can just use `.values()`. All the different accessors are needed in C++ because the C++ collections library is super impoverished.